### PR TITLE
HS-1384: use OK-HTTP instead of NETTY for GRPC connections to elimina…

### PR DESCRIPTION
**Pull request by artnaseef**
…te connection leaking.

- Update ingress-nginx version to v1.7.0
  - Updated cluster role permissions to match
- Use OK HTTP GRPC client channel code instead of NETTY GRPC client code
  - Almost completely eliminates lingering connections after a GOAWAY is received from the server (nginx)
  - Cleans up the lingering connections so we don't leak over time
  - Includes OK HTTP method of configuring SSL which is notably different than the NETTY method
- Basic build tools:
  - update helm chart values to add new certificates and SSL configuration
  - Use image "horizon-stream-minion-certificate-verifier:local-basic"

## Jira link(s)
- https://issues.opennms.org/browse/HS-xx

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [x] Documentation has been updated as necessary.
* [x] Notify devops of changes to the Charts

### CODE REVIEW

File: build-tools/basic/compile.sh [hunk 1]
```diff
@@ -38,3 +38,4 @@ mvn clean install -P rapid-build -DskipTests=true -f notifications
 mvn clean install -P rapid-build -DskipTests=true -f metrics-processor
 mvn clean install -P rapid-build -DskipTests=true -f events
 mvn clean install -P rapid-build -DskipTests=true -f datachoices
+mvn clean install -P rapid-build -DskipTests=true -f minion-certificate-manager
```
Comment (splatch) [inline build-tools/basic/compile.sh:41]: As far I can see this module is listed in line 32 of this script

File: build-tools/basic/mk-docker-images.sh [hunk 2]
```diff
@@ -86,6 +86,12 @@ time {
 	echo "==="
 	mvn -f datachoices install -Djib.container.creationTime=USE_CURRENT_TIMESTAMP -Dapplication.docker.image=opennms/horizon-stream-datachoices:local-basic
 
+	echo ""
+	echo "==="
+	echo "=== CERTIFICATE MANAGER IMAGE"
+	echo "==="
+	mvn -f minion-certificate-manager install -Djib.container.creationTime=USE_CURRENT_TIMESTAMP -Dapplication.docker.image=opennms/horizon-stream-minion-certificate-manager:local-basic
```
Comment (splatch) [inline build-tools/basic/mk-docker-images.sh:93]: See line 39 of script.

File: build-tools/basic/push-images-to-kind.sh [hunk 3]
```diff
@@ -51,6 +51,7 @@ kind load docker-image "opennms/horizon-stream-datachoices:local-basic"
 # 		"opennms/horizon-stream-inventory:local-basic" \
 # 		"opennms/horizon-stream-keycloak:local-basic" \
 # 		"opennms/horizon-stream-metrics-processor:local-basic" \
+# 		"opennms/horizon-stream-minion-certificate-manager:local-basic" \
```
Comment (splatch) [inline build-tools/basic/push-images-to-kind.sh:54]: See line 36.

**Review by splatch — CHANGES_REQUESTED** [review]
Please update shell scripts as they do contain duplicate build calls which will impact build time.

File: build-tools/basic/compile.sh (hunk 1 above)
Comment (artnaseef) [reply build-tools/basic/compile.sh:41]: Good catch

File: build-tools/basic/push-images-to-kind.sh (hunk 3 above)
Comment (artnaseef) [reply build-tools/basic/push-images-to-kind.sh:54]: This one isn't actually a duplicate - notice it is commented-out.

File: build-tools/basic/mk-docker-images.sh (hunk 2 above)
Comment (artnaseef) [reply build-tools/basic/mk-docker-images.sh]: Another good catch.

**Review by splatch — APPROVED** [review]

**Pull request merged**